request.js

```diff
@@ -1413,6 +1413,28 @@ Request.prototype.jar = function (jar) {
   return self
 }
 
+Request.prototype.cloneJar = function (cookieJar) {
+  var self = this
+  var newJar, cookieString, cookieStore, domains, domainPaths, cookieUrl
+
+  newJar = self.jar()
+  cookieStore = cookieJar._jar.store.idx
+  domains = Object.keys(cookieStore)
+
+  domains.forEach(function (domain) {
+    domainPaths = Object.keys(cookieStore[domain])
+
+    domainPaths.forEach(function (domainPath) {
+      cookieUrl = 'http://' + domain + domainPath
```

Comment:
That was my only doubt, I tested it with https/http, thing is to my knowledge cookies only require domain, not protocol so this was added only because otherwise `getCookieString` wouldn't parse correctly.